### PR TITLE
feat(arithmetic): #1347 #1378 ➗ truediv

### PR DESCRIPTION
**Pull request by cmp0xff**
- [x] Addresses #1347 and #1378
- [x] Tests added

### CODE REVIEW

File: tests/series/arithmetic/bool/test_truediv.py [hunk 1]
```diff
@@ -41,13 +50,15 @@ def test_truediv_py_scalar() -> None:
     )
 
     if TYPE_CHECKING_INVALID_USAGE:
-        left.div(b)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
+        # TODO: python/mypy#20061
```
Comment (cmp0xff) [inline tests/series/arithmetic/bool/test_truediv.py:53]: Note that python/mypy#20061 arises at a few places

File: tests/indexes/arithmetic/timedelta/test_truediv.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    check(assert_type(left / d, "pd.Index[float]"), pd.Index, np.floating)
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        _10 = b / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _11 = i / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _12 = f / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _13 = c / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _14 = s / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+    # TODO: pandas-dev/pandas#62712
+    # check(assert_type(d / left, "pd.Index[float]"), pd.Index, np.floating)
```
Comment (cmp0xff) [inline tests/indexes/arithmetic/timedelta/test_truediv.py]: Note that #pandas-dev/pandas#62712 also arises. But I think `Index[Timedelta]` is really an edge case.

File: pandas-stubs/core/indexes/base.pyi [hunk 3]
```diff
@@ -902,18 +910,166 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
...
+        other: (
+            Just[complex]
+            | Sequence[Just[complex]]
+            | np_ndarray_complex
+            | Index[complex]
+        ),
+    ) -> Index[complex]: ...
+    @overload
+    def __truediv__(
+        self: Index[Timedelta],
```
Comment (Dr-Irv) [inline pandas-stubs/core/indexes/base.pyi]: Really would prefer that we don't support operations on `Index[Timedelta]` because that is a `TimedeltaIndex`, so if somehow our stubs are producing `Index[Timedelta]`, that should be fixed.

File: pandas-stubs/core/indexes/timedeltas.pyi [hunk 4]
```diff
@@ -40,6 +39,20 @@ from pandas._typing import (
     num,
 )
 
+_NUM_FACTOR: TypeAlias = Just[int] | Just[float] | np.integer | np.floating
+_NUM_FACTORS: TypeAlias = (
```
Comment (Dr-Irv) [inline pandas-stubs/core/indexes/timedeltas.pyi]: Easier to read if you called it `_NUM_FACTOR_SEQ` .

File: pandas-stubs/core/indexes/timedeltas.pyi [hunk 5]
```diff
@@ -40,6 +39,20 @@ from pandas._typing import (
...
+    _NUM_FACTOR
+    | Sequence[_NUM_FACTOR]
+    | np_ndarray_anyint
+    | np_ndarray_float
+    | Index[int]
+    | Index[float]
+)
+_DT_FACTOR: TypeAlias = dt.timedelta | np.timedelta64 | Timedelta
+_DT_FACTORS: TypeAlias = (
```
Comment (Dr-Irv) [inline pandas-stubs/core/indexes/timedeltas.pyi]: same comment as above

File: tests/series/arithmetic/bool/test_truediv.py (hunk 1 above)
Comment (Dr-Irv) [reply tests/series/arithmetic/bool/test_truediv.py:53]: ok

File: tests/indexes/arithmetic/float/test_truediv.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    """Test pd.Index[float] / pandas scalars"""
+    s, d = pd.Timestamp(2025, 9, 24), pd.Timedelta(seconds=1)
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        _00 = left / s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _01 = left / d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        _10 = s / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+    check(assert_type(d / left, "pd.Index[pd.Timedelta]"), pd.Index, pd.Timedelta)
```
Comment (Dr-Irv) [inline tests/indexes/arithmetic/float/test_truediv.py]: This should be `TimedeltaIndex` as the result.

File: tests/indexes/arithmetic/int/test_truediv.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    """Test pd.Index[int] / pandas scalars"""
+    s, d = pd.Timestamp(2025, 9, 24), pd.Timedelta(seconds=1)
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        _00 = left / s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _01 = left / d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        _10 = s / left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+    check(assert_type(d / left, "pd.Index[pd.Timedelta]"), pd.Index, pd.Timedelta)
```
Comment (Dr-Irv) [inline tests/indexes/arithmetic/int/test_truediv.py]: same comment as above

File: tests/indexes/arithmetic/timedelta/test_truediv.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+)
+
+from tests import (
+    TYPE_CHECKING_INVALID_USAGE,
+    check,
+)
+
+
+@pytest.fixture
+def left() -> "pd.Index[pd.Timedelta]":
```
Comment (Dr-Irv) [inline tests/indexes/arithmetic/timedelta/test_truediv.py]: This should be returning `TimedeltaIndex`

File: tests/indexes/arithmetic/timedelta/test_truediv.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    """Left operand"""
+    # pandas-dev/pandas#62524: An index of Python native timedeltas can be
+    # produced, instead of a TimedeltaIndex, hence this test file
```
Comment (Dr-Irv) [inline tests/indexes/arithmetic/timedelta/test_truediv.py]: While this can be produced, I think the stubs should not allow it (if that's possible).

File: tests/indexes/arithmetic/timedelta/test_truediv.py (hunk 2 above)
Comment (Dr-Irv) [reply tests/indexes/arithmetic/timedelta/test_truediv.py]: Is there a way that we can just have the stubs not accept operations on `Index[Timedelta]` , even though some work at runtime?  Then if someone actually creates that, we suggest a better way of coding that uses `TimedeltaIndex`

File: tests/indexes/arithmetic/test_truediv.py [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+
+def test_truediv_path(tmp_path: Path) -> None:
+    """Test pd.Index / path object.
+
+    Also GH 682."""
+    fnames = pd.Index(["a.png", "b.gz", "c.txt"])
+
+    if PD_LTE_23:
+        # Bug in 3.0 https://github.com/pandas-dev/pandas/issues/61940 (pyarrow.lib.ArrowInvalid)
```
Comment (Dr-Irv) [inline tests/indexes/arithmetic/test_truediv.py]: that issue is closed, so can you see if it is still a problem?  The issue was about series, but this is testing index.

Comment (cmp0xff) [reply tests/indexes/arithmetic/test_truediv.py]: 70e0f2a3ecfb9831eff29e44bb11b795fb91129a

File: tests/indexes/arithmetic/timedelta/test_truediv.py (hunk 8 above)
Comment (cmp0xff) [reply tests/indexes/arithmetic/timedelta/test_truediv.py]: At runtime, we actually get `Index([0:00:01], dtype='object')`

File: tests/indexes/arithmetic/timedelta/test_truediv.py (hunk 9 above)
Comment (cmp0xff) [reply tests/indexes/arithmetic/timedelta/test_truediv.py]: `pd.Index[pd.Timedelta]` can easily be produced by `mul` or `truediv` at the moment. I think the pandas core developers should decide whether to allow it.

File: pandas-stubs/core/indexes/timedeltas.pyi (hunk 5 above)
Comment (cmp0xff) [reply pandas-stubs/core/indexes/timedeltas.pyi]: ca7d5a9f0d54d70b77dba8c43ea1ce5491d0ba3e

File: pandas-stubs/core/indexes/timedeltas.pyi (hunk 4 above)
Comment (cmp0xff) [reply pandas-stubs/core/indexes/timedeltas.pyi]: ca7d5a9f0d54d70b77dba8c43ea1ce5491d0ba3e

**Comment by Dr-Irv** [comment]
> I cannot delete it any further.
> 
> ```powershell
> > git rm .\tests\indexes\arithmetic\timedelta
> fatal: pathspec '.\tests\indexes\arithmetic\timedelta' did not match any files
> ```
> 
> I think `git` does not keep empty folders.

OK.  I approved and merged and the folder is gone.  But the review process indicated it was still there.  Hence the confusion
